src/generation/level_matrix.py

```diff
@@ -0,0 +1,125 @@
+import random
+from datetime import datetime
+import src.generation.shape_template
+
+
+class LevelMatrix:
+    def __init__(self, matrix, known_shapes):
+        self.matrix = matrix.copy()
+        self.directions = ((-1, -1), (0, -1), (1, -1), (-1, 0), (1, 0), (-1, 1), (0, 1), (1, 1))
```

Comment:
Maybe make this global somewhere. It's generic and should not take space for each instance